On top of this patch we include a few more redirecting versions of
the rtld-* routines in multiarch, this is needed because of the
sysd-rules which try to build from multiarch first and will even
use memset.S from multiarch to buidl rtld-memset.os, which results
in SSE register usage in rtld which we can't allow because of
lazy binding (though in some cases we use RTLD_PREPARE_FOREIGN_CALL
and others to wrap such calls).

commit 747ef469ffc9c9179ca9d76854167925b4e44346
Author: Siddhesh Poyarekar <siddhesh@redhat.com>
Date:   Sat Jun 15 00:09:26 2013 +0530

    Add rtld-memset.S for x86_64
    
    Resolves: BZ #15627
    
    Add an assembler version of rtld-memset to avoid using SSE registers.

Index: glibc-2.17-c758a686/sysdeps/x86_64/rtld-memset.S
===================================================================
--- /dev/null
+++ glibc-2.17-c758a686/sysdeps/x86_64/rtld-memset.S
@@ -0,0 +1,37 @@
+/* memset implementation for the dynamic linker.  This is separate from the
+   libc implementation to avoid writing to SSE registers.
+   Copyright (C) 2013 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#include <sysdep.h>
+#include "asm-syntax.h"
+
+
+	.text
+/* void *memset (void *dest, char c, size_t count)
+   dest	 => %rdi
+   c	 => %rsi
+   count => %rdx  */
+ENTRY (memset)
+	mov	%rdx, %rcx
+	movzbl	%sil, %eax
+	mov	%rdi, %rdx
+	rep	stosb
+	mov	%rdx, %rax
+	ret
+END (memset)
+libc_hidden_builtin_def (memset)
Index: glibc-2.17-c758a686/sysdeps/x86_64/rtld-memset.c
===================================================================
--- glibc-2.17-c758a686.orig/sysdeps/x86_64/rtld-memset.c
+++ /dev/null
@@ -1 +0,0 @@
-#include <string/memset.c>
Index: glibc-2.17-c758a686/sysdeps/x86_64/multiarch/rtld-memset.S
===================================================================
--- /dev/null
+++ glibc-2.17-c758a686/sysdeps/x86_64/multiarch/rtld-memset.S
@@ -0,0 +1 @@
+#include "../rtld-memset.S"
Index: glibc-2.17-c758a686/sysdeps/x86_64/multiarch/rtld-strchr.S
===================================================================
--- /dev/null
+++ glibc-2.17-c758a686/sysdeps/x86_64/multiarch/rtld-strchr.S
@@ -0,0 +1 @@
+#include "../rtld-strchr.S"
